File: 2026-bogota-agent-sdk/.claude/skills/crear-agente-sdk/references/ejemplos/python/01_query.py
"""01 · query() — un prompt entra, una respuesta sale.

Sin tools. El modelo solo puede contestar con texto, así que a la pregunta del
taller responde que no puede ver tu correo. Ese "no puedo" es el punto de
partida: los archivos 02 a 08 le van dando las piezas que le faltan.

Fíjate en que el script corre y termina. Un agente no necesita un chat.

    uv run python 01_query.py
"""

import asyncio

from claude_agent_sdk import ClaudeAgentOptions, query

from mostrar import corto

PREGUNTA = "Revisa mis correos de la última semana y dime cuándo es mi próximo vuelo."

SYSTEM_PROMPT = """Eres un asistente personal. Respondes en español, corto y con datos.
Usa tus tools y skills; no adivines fechas ni reservas."""


async def main():
    opciones = ClaudeAgentOptions(
        model="claude-sonnet-5",
        system_prompt=SYSTEM_PROMPT,
        tools=[],                       # ninguna tool: solo texto
        setting_sources=[],             # nada del disco: ni skills ni CLAUDE.md de quien corre esto
        strict_mcp_config=True,         # ni los conectores de claude.ai de tu cuenta
        env={"CLAUDE_CODE_DISABLE_AUTO_MEMORY": "1"},   # ni tu memoria personal
        max_turns=2,
        max_budget_usd=0.05,
        stderr=lambda _: None,
    )

    print(f"PREGUNTA  {PREGUNTA}\n")
    async for m in query(prompt=PREGUNTA, options=opciones):
        # El stream trae varios mensajes; aquí solo nos interesa el último.
        if type(m).__name__ == "ResultMessage":
            print(f"RESPUESTA {corto(m.result or '', 400)}\n")
            print(f"RECIBO    USD {m.total_cost_usd:.4f} · {m.duration_ms / 1000:.1f}s")


asyncio.run(main())   # en un notebook esta línea falla: allí se usa `await main()`

File: 2026-bogota-agent-sdk/.claude/skills/crear-agente-sdk/references/ejemplos/typescript/01_query.ts
// 01 · query(): entra un prompt, sale una respuesta. El programa termina solo.
// Sin ninguna tool. El agente no puede mirar tu correo, y lo dice.
import "./entorno.ts"; // carga ANTHROPIC_API_KEY de .env
import { query } from "@anthropic-ai/claude-agent-sdk";

const SISTEMA =
  "Eres un asistente personal. Respondes en español, corto y con datos. " +
  "Usa tus tools y skills; no adivines fechas ni reservas.";

const PREGUNTA = "Revisa mis correos de la última semana y dime cuándo es mi próximo vuelo.";

console.log(`Tú: ${PREGUNTA}\n`);

const conversacion = query({
  prompt: PREGUNTA,
  options: {
    model: "claude-sonnet-5",
    systemPrompt: SISTEMA,
    tools: [], // ninguna tool disponible: el agente solo puede hablar
    settingSources: [], // no lee ~/.claude ni el .claude del proyecto
    strictMcpConfig: true, // no entran los conectores de claude.ai
    maxTurns: 2,
    maxBudgetUsd: 0.05,
    env: { ...process.env, CLAUDE_CODE_DISABLE_AUTO_MEMORY: "1" },
  },
});

// El stream se recorre entero: salir con break se deja mensajes sin leer.
for await (const mensaje of conversacion) {
  if (mensaje.type === "assistant") {
    for (const bloque of mensaje.message.content) {
      if (bloque.type === "text") console.log(`Agente: ${bloque.text}`);
    }
  }
  if (mensaje.type === "result" && mensaje.subtype === "success") {
    console.log(`\nCosto: $${mensaje.total_cost_usd.toFixed(4)}`);
  }
}
